src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResource.kt

```diff
@@ -72,7 +74,14 @@ class AppointmentMappingResource(private val mappingService: AppointmentMappingS
     @RequestBody @Valid
     createMappingRequest: AppointmentMappingDto,
   ) =
-    mappingService.createMapping(createMappingRequest)
+    try {
+      mappingService.createMapping(createMappingRequest)
+    } catch (e: DuplicateKeyException) {
+      throw DuplicateMappingException(
```

Comment:
seems slightly odd here to not pass the cause `e` to the new exception too

Reply:
I've embedded the cause in the message, that seems enough, and convenient for the client end